network.go

```diff
@@ -1184,6 +1290,27 @@ func isPhysicalIface(ifaceName string) (bool, error) {
 	return true, nil
 }
 
+// isVhostuserIface checks if an interface is a dummy placeholder
+// for a vhost-user socket
+func isVhostuserIface(netInfo NetworkInfo) (bool, string, error) {
+	if netInfo.Iface.Name == "lo" {
+		return false, "", nil
+	}
+
+	// for now, check for socket file existence at known location.
+	// Identify this be searching for a file in a path which contains
+	// the interfaces IP address.  ie:
+	//  /tmp/vhostuser_<ipv4 address>/vhu.sock
+	for _, addr := range netInfo.Addrs {
+		socketPath := fmt.Sprintf("/tmp/vhostuser_%s/vhu.sock", addr.IPNet.IP)
```

Comment:
I think we had a discussion some time ago, about not putting things into /tmp since it can be accessed by anyone on the system. It'd be safer to use a path like `/var/run/virtcontainers/pods/<podID>/vhostuser_<IP>/vhu.sock`.

Reply:
@sboeuf  

Let me see what I can do here.  

This is actually not a function of the runtime, but a function of where the CNI and/or CNM plugin places these.  Let me see what I can change --  I'm consuming these sockets, not creating them.

Reply:
Oh I see, then ignore this comment.

Reply:
This ``/tmp/vhostuser_%s/vhu.sock`` is based on the CNI/CNM plugin we use. Ideally this should be configurable. vhostuser is very flexible in that the socket can be in its own dir or all sockets can also be created in the same directory. We can make an assumption that there is a 
- single socket / dir
What happens when there are two vhost user networks (say VPP with one L2 switch for inbound and L2 switch for outbound). 

When there are unique IP addresses across networks this scheme will work.

The only issue I can see is if there are 2 networks and 2 containers. But both the networks have the same IP, but each container connects to only one network. This is a corner case which we can ignore now as this will not show up in the case of kubernetes.

Reply:
So in summary let us make this search location configurable.